docs/data/joy/customization/dark-mode/ModeToggle.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import { CssVarsProvider, useColorScheme } from '@mui/joy/styles';
+import Box from '@mui/joy/Box';
+import Button from '@mui/joy/Button';
+
+const useEnhancedEffect =
+  typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
+
+function ModeSwitcher() {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    return null;
+  }
+  return (
+    <Button
+      variant="outlined"
+      color="neutral"
+      onClick={() => setMode(mode === 'dark' ? 'light' : 'dark')}
+    >
+      {mode === 'dark' ? 'Turn light' : 'Turn dark'}
+    </Button>
+  );
+}
+
+export default function ModeToggle() {
+  // the `node` is used for attaching CSS variables to this demo, you might not need it in your application.
```

Comment:
Scroll overflow, I would recommend shortening:

<img width="812" alt="Screenshot 2022-11-07 at 02 15 20" src="https://user-images.githubusercontent.com/3165635/200207271-1953e8a9-e0d0-4701-bd88-9ca36ed77037.png">

```suggestion
  // the `node` is used for attaching CSS variables to this demo, you might not need
  // it in your application.
```

*x the other comments*